crypt.go

```diff
@@ -439,7 +440,36 @@ func ActivateVolumeWithMultipleKeyData(volumeName, sourceDevicePath string, keys
 		return errors.New("nil kdf")
 	}
 
-	s := newActivateWithKeyDataState(volumeName, sourceDevicePath, options.KeyringPrefix, options.Model, keys, authRequestor, kdf, options.PassphraseTries)
+	var keyContexts []*keyDataContext
+	for _, key := range keys {
+		keyContexts = append(keyContexts, &keyDataContext{KeyData: key, slot: luks2.AnySlot})
+	}
+
+	view, err := newLUKSView(sourceDevicePath, luks2.LockModeBlocking)
+	if err != nil {
+		fmt.Fprintf(osStderr, "secboot: Cannot obtain LUKS2 header view: %v\n", err)
+	} else {
+		tokens := view.KeyDataTokensByPriority()
+		for _, token := range tokens {
+			if token.Data == nil {
+				// Skip uninitialized token
+				continue
+			}
+
+			r := &LUKS2KeyDataReader{
+				name:   sourceDevicePath + ":" + token.Name(),
+				Reader: bytes.NewReader(token.Data)}
+			kd, err := ReadKeyData(r)
+			if err != nil {
+				fmt.Fprintf(osStderr, "secboot: Cannot read keydata from token %s: %v\n", token.Name(), err)
```

Comment:
I've updated the messages added by this PR.

For context, I'd capitalised other messages logged to stderr because they aren't wrapped by other errors and capitalisation in wrapped errors looks weird, but I can update the other messages in another PR.